corporate/lib/stripe.py

```diff
@@ -3569,13 +3625,15 @@ def update_or_create_customer(
             return customer
 
     @override
+    @transaction.atomic
     def do_change_plan_type(
         self, *, tier: Optional[int], is_sponsored: bool = False
     ) -> None:  # nocoverage
         # This function needs to translate between the different
         # formats of CustomerPlan.tier and RealmZulipServer.plan_type.
         if is_sponsored:
             plan_type = RemoteZulipServer.PLAN_TYPE_COMMUNITY
+            self.add_customer_to_community_plan()
```

Comment:
There's probably a future refactor we can do to get rid of the `is_sponsored` parameter to this function and just accept the `tier` parameter with `TIER_SELF_HOSTED_COMMUNITY` as the parameter.